### DISCUSSION TITLE
Migrate search mode from Priority to Mincost

### DISCUSSION THREAD

**Discussion by maxware0715** [frePPLe/frepple#646] [General]
Assumed that I have a route.
In specific operation A, defined multiple input items in operation materials with different priority.
All these items have item suppliers in their upstream operations.

If I defined cost in item suppliers and change 'search' in operation materials from 'PRIORITY' to 'MINCOST'
expected below planning process happened:
   1. if no purchased order need, consumed confirmed manufacturing orders in order of priority
   2. 'name' in operation materials still defined alternative relationship
   3. if purchased order need, produced purchase order with minimum cost

briefly speaking, would 'MINCOST' keeps 'PRIORITY' behaviors if not conflicts?
or it is a must to convert alternative items defined in operation materials to multiple alternative operations?

**Comment by jdetaeye** [comment]
The behavior I expect to see matches your description.   

I would clarify/rephrase "1. if no purchased order need, consumed confirmed manufacturing orders in order of priority" as follows. If there is stock or confirmed purchase orders on any of the alternate materials, frepple will consume that supply first before proposing a new purchase order.   This use-up search uses the priority indeed.

Once the existing supply is all used up, the algorithm will in step 3 choose purchasing from the cheapest supplier that can meet the requirement date.   Note that if the cheapest supplier has a very long lead time, the algorithm will buy from more expensive suppliers that have a shorter lead time.

**Reply by maxware0715** [reply]
Thank you.
I guess the clarify you make remain the same when search='PRIORITY'?

My target is to consume all stock and confirmed item first, then create new purchase orders from highest priority route.
Currently we achieve this by add a long time(365 days) in itemsupplier.leadtime
And I am doing experiments for finding a solution without manually manipulate the time.

My version is 6.25. Please remind me if any difference compares with the latest version.

**Reply by jdetaeye** [reply]
>> My version is 6.25. 

That's pretty ancient, and many things have evolved since - too much too remember.   Always try with the latest version...
